agent/consul/health_endpoint.go

```diff
@@ -21,12 +22,23 @@ type HealthGRPCAdapter struct {
 	Health
 }
 
-func (h *HealthGRPCAdapter) Stream(in *TestRequest, stream Health_StreamServer) error {
+func (h *HealthGRPCAdapter) Subscribe(req *stream.SubscribeRequest, server stream.Consul_SubscribeServer) error {
+	if !strings.HasPrefix(in.Key, "health/service-nodes/") {
+		return fmt.Errorf("only health/service-nodes requests are supported")
+	}
+
```

Comment:
- switch on topic
  - separate snapshot method for each topic
  - handler must
    - load current snapshot and iterate records
    - convert memDB record(s) into event format needed
    - send them on a chan, blocking but with context check from the RPC stream.